website/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx

```diff
@@ -0,0 +1,317 @@
+  const dataLocale = useApplicationContext((context) =>
+    applicationContext.dataLocale
+  );
+  render() {
+    return (
+      <div>
+        <h1>Current data locale: {dataLocale}</h1>
+      </div>
+    );
+  }
+}
+
+export default DisplayLocale;
+```
+
+### withApplicationContext
+
+A higher-order component (HOC) that allows to access the `ApplicationContext` and selectively pick the necessary properties. This is the equivalent of the React hook `useApplicationContext`.
+
+```jsx highlightLines="13-15"
+import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+class DisplayLocale extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Current data locale: {this.props.dataLocale}</h1>
+      </div>
+    );
+  }
+}
+
+export default withApplicationContext((applicationContext) => ({
+  dataLocale: applicationContext.dataLocale
+}))(DisplayLocale);
+```
+
+# Project image settings
+
+Product images can be uploaded to the commercetools platform or can be referenced from external sources.
+
+By default, images referenced from external sources are not displayed in the Merchant Center. This is primarily due to avoid possible performance issues in case the size of those images is big.
+
+In order to show images referenced from external sources, you need to define a configuration to rewrite the URL of the images to versions of the image that has a smaller size. You can configure that in the Merchant Center **Settings > Project Settings > Miscellaneous**.
+
+This configuration can be fetched from your Custom Application using the following components.
+
+## Usage
+
+### ProjectExtensionProviderForImageRegex
+
+This is the React context provider that loads the image regex configuration and expose it in the React context so that children can access the configuration.
+
+<Info>
+
+This component is required to be defined in a parent component where children of this component need to access the configuration.
+
+</Info>
+
+#### Properties
+
+| Props                           | Type                       | Required | Values            | Default   | Description                                                              |
+| ------------------------------- | -------------------------- | :------: | ----------------- | --------- | -------------------------------------------------------------------------|
+| `children`                      | `ReactNode`                |    ✅    | -                 | -         | Components that should be rendered within the scope of this provider.    |
+| `skip`                          | `boolean`                  |    -     | -                 | false     | Disables loading images configuration.                                   |
+
+
+### GetProjectExtensionImageRegex
+
+Use this component to access the project images configuration, using a `render` prop function.
+
+```jsx
+function MyComponent() {
+  return (
+    <GetProjectExtensionImageRegex
```

Comment:
At some point we should likely also offer a `useProjectExtensionImageRegex` and build the render-prop based component and HoC upon it maybe.

Reply:
Yes we probably should

Reply:
I created a [new GitHub issue](https://github.com/commercetools/merchant-center-application-kit/issues/2571) to implement that hook as a follow up.